Nef_S2/include/CGAL/Nef_S2/Sphere_circle.h

```diff
@@ -134,6 +140,10 @@ of splitting |\Mvar| at the $x$-$y$-coordinate plane if |\Mvar|
 is not part of it. Otherwise |\Mvar| is split at the
 $x$-$z$-coordinate plane.}*/
 
+private:
+struct Private_tag{};
```

Comment:
The name must be improved or we must even come up with a better design than such a tag class. @sloriot any idea? 
If we keep it, it should have a name telling us what we want to achieve.

Reply:
It seems to be a name used elsewhere: https://github.com/CGAL/cgal/blob/b575892a164d83e6d47a960839f86ae24dca4114/STL_Extension/include/CGAL/Object.h#L135

Reply:
@lrineau I wasn't intending to imply a 'member access modifier' by the name 'Private' in the name of the tag. Rather, it's just an implementation detail really, so perhaps `CGAL::internal_tag` might be more appropriate. Like I say I think I chose `Private_tag` because it was used elsewhere in the CGAL codebase. 
@afabri I don't like the design either, but it is a neat workaround for what I consider a limitation of C++ when you need two copy constructors that take the same parameter type, but with different behaviour.

Reply:
Anyway, a name like `Skip_the_test` or `Assume_d_equal_0` is more informative than something related to "private" or "internal".

Reply:
@lrineau I've changed it to `Assume_d_equal_0` for now. I mulled over `Assume_origin_intercept`, or just `Intercepts_origin`...